src/main/java/strategy/PurchaseOperation.java

```
@@ -0,0 +1,18 @@
+package strategy;
+
+import db.Storage;
+import model.FruitTransaction;
+
+public class PurchaseOperation implements OperationHandler {
+
+    @Override
+    public void apply(FruitTransaction transaction) {
+        String fruit = transaction.getFruit();
+        int current = Storage.getQuantity(fruit);
+        int newBalance = current - transaction.getQuantity();
+        if (newBalance < 0) {
+            throw new RuntimeException("Negative balance for fruit: " + fruit);
```

When the new balance would be negative you throw `new RuntimeException("Negative balance for fruit: " + fruit)`. While this enforces the purchase handling rule, the requirements also say: "Do not ignore exceptions; rethrow as RuntimeException with informative messages and original cause." Provide a more informative message (include `fruit`, attempted `quantity`, `current` balance and ideally transaction identifier) so callers can quickly locate the problematic CSV row/transaction.